tests/privacy_scenario_3_test.py

```diff
@@ -0,0 +1,290 @@
+
+    # before last recorded block number (rbn4) is made irreversible, verify
+    # node1 and node2 still stay in sync
+    apiNode1_head = apiNode1.getHeadBlockNum()
+    Print("API Node1 head block = {}".format(apiNode1_head))
+    apiNode2_head = apiNode2.getHeadBlockNum()
+    Print("API Node2 head block = {}".format(apiNode2_head))
+    apiNode1_InSync = False
+    apiNode2_InSync = False
+    while prod1.getIrreversibleBlockNum() < trans_block_num:
+        temp = apiNode1.getHeadBlockNum()
+        if apiNode1_head < temp:
+            apiNode1_InSync = True
+            apiNode1_head = temp
+        temp = apiNode2.getHeadBlockNum()
+        if apiNode2_head < temp:
+            apiNode2_InSync = True
+            apiNode2_head = temp
+        prod1.waitForNextBlock(sleepTime=0.5)
+    
+    Print("API Node1 head = {}".format(apiNode1_head))
+    Print("API Node2 head = {}".format(apiNode2_head))
+
+    assert trans_block_num < apiNode2_head
+    assert apiNode1_InSync
+    Print("OK: API Node1 head {} advanced past transaction block {}".format(apiNode1_head, trans_block_num))
+    assert trans_block_num < apiNode1_head
+    assert apiNode2_InSync
+    Print("OK: API Node2 head {} advanced past transaction block {}".format(apiNode2_head, trans_block_num))
+
+    # after last recorded block number (rbn4) is made irreversible, verify
+    # node1 and node2 do not advance their LIB to rbn4 and do not continue to receive blocks
+    waitSec = 6
+    Print("Waiting for {} seconds to ensure API nodes LIB doesn't move".format(waitSec))
+    time.sleep(waitSec)
+
+    apiNode1_lib = apiNode1.getIrreversibleBlockNum()
+    Print("API Node1 LIB = {}".format(apiNode1_lib))
+    assert apiNode1_lib < trans_block_num
+    apiNode2_lib = apiNode2.getIrreversibleBlockNum()
+    Print("API Node2 LIB = {}".format(apiNode2_lib))
+    assert apiNode2_lib < trans_block_num
+
+    # add node1 back to security group *
+    trans = securityGroup.editSecurityGroup(addNodes=[apiNode1])
+    # record block number (rbn5) transaction was added in
+    trans_block_num = prod1.getBlockNumByTransId(trans["transaction_id"], blocksAhead=3)
+    assert trans_block_num is not None
+
+    apiNode1_head = apiNode1.getHeadBlockNum()
+    apiNode2_head = apiNode2.getHeadBlockNum()
+    # before last recorded block number (rbn5) is made irreversible, verify
+    # node1 and node2 do not receive blocks
+    breakTimeout = 100
+    while breakTimeout > 0:
+        curApiNode1_head = apiNode1.getHeadBlockNum()
+        Print("API Node1 head = {}".format(curApiNode1_head))
+        curApiNode2_head = apiNode2.getHeadBlockNum()
+        Print("API Node2 head = {}".format(curApiNode2_head))
+        prod1LIB = prod1.getIrreversibleBlockNum()
+        if prod1LIB >= trans_block_num:
+            break
+        
+        assert curApiNode1_head == apiNode1_head
+        assert curApiNode2_head == apiNode2_head
+        
+        prod1.waitForNextBlock(sleepTime=0.5)
```

Comment:
assert ...wait.. != None

Reply:
I found better solution, it does have `errorContext` string parameter and if it is not `None` it asserts